compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -549,6 +551,7 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
         tcx,
         body,
         &[
+            &large_enums::EnumSizeOpt { discrepancy: 128 },
```

Comment:
This pass replaces direct assignments by indirect copies through pointers.
As no other pass does any sort of alias analysis, this pass will confuse them.
Could you push it later in the pipeline, so it only benefits codegen?

Reply:
I think I've pushed it deeper, but not sure if this is where you intended.

Reply:
I meant really at the end. After deduplicateblocks for instance. 
That way codegen sees the transformation, while MIR opts see a standard assignment (that they can try to eliminate on their own).

Reply:
Can you try removing the `-Zmir-opt-level` argument in the test? The `unit-test` comment should be enough to test this pass in isolation.

Reply:
I removed the `--Zmir-opt-level` argument,
but due to the `if (!sess.opts.unstable_opts.unsound_mir_opts) || sess.mir_opt_level() < 3 { return; }`, it still doesn't seem to be run. Is there documentation for the `unit-test` comment to make sure I'm doing it correctly?